cores/portduino/linux/LinuxHardwareSPI.cpp

```diff
@@ -72,6 +76,7 @@ class LinuxSPIChip : public SPIChip, private PosixFile {
       assert (ioctl(SPI_IOC_WR_MAX_SPEED_HZ, &defaultSpeed) >= 0);
     }
 };
+std::map<std::string, std::shared_ptr<void>> SPI_map;
```

Comment:
use **unordered_map** here for more efficiency, bc it does not use balanced trees. map is only used when the order in which the elements were created is important and used later (e.g. for printing all elements).

Reply:
Also here I in this lower layer I would never use strings as a hash index. Imagine this function is called hundred times per second. Why not use X.Y (=spi_host) directly as an index which has most efficiency.